packages/flutter/lib/src/material/carousel.dart

```diff
@@ -490,6 +634,505 @@ class _RenderSliverFixedExtentCarousel extends RenderSliverFixedExtentBoxAdaptor
   ItemExtentBuilder? get itemExtentBuilder => _buildItemExtent;
 }
 
+/// A sliver that arranges its box children in a linear array, constraining them
+/// to specific weights determined by the [weights] property.
+///
+/// _To learn more about slivers, see [CustomScrollView.slivers]._
+///
+/// This sliver arranges its children in a line along the main axis, starting
+/// at offset zero without gaps. Each child is constrained to its corresponding
+/// weight along the main axis and to the [SliverConstraints.crossAxisExtent]
+/// along the cross axis.
+///
+/// While scrolling, the extent (size) of each visible item changes dynamically
+/// based on the scrolling progress.  As the first visible item scrolls completely
+/// off-screen, the next item becomes the first visible item, and has the same
+/// size as the previously first item. The rest of the visible items maintain
+/// their relative layout.
+///
+/// For example, the layout weights is [1, 6, 1]. The length of [weights] array
+/// indicates three items should be visible at a time. The layout of these items
+/// would be:
+///  * First item: Extent is (1 / (1 + 6 + 1)) * viewport extent.
+///  * Second item: Extent is (6 / (1 + 6 + 1)) * viewport extent.
+///  * Third item: Extent is (1 / (1 + 6 + 1)) * viewport extent.
+///
+/// Assuming a viewport extent of 800 in the main axis and the first item is
+/// item 0, there would be three visible items with extents of 100, 600, and 100.
+/// As item 0 scrolls off-screen, the extent of item 1 smoothly decreases from 600
+/// to 100. For instance, if item 0 is 30% off-screen, item 1 should have decreased
+/// its size to 30% of the difference from 600 to 100; its extent would be
+/// 600 - 0.3 * (600 - 100). Similarly, item 2's extent would increase from 100
+/// to 600, becoming 100 + 0.3 * (600 - 100).
+///
+/// As the initially visible items change size during scrolling, item 3 enters
+/// the view to fill the remaining space. Its extent starts at a minimum of
+/// [shrinkExtent] (or 0 if [shrinkExtent] is not provided) and gradually
+/// increases to match the extent of the last visible item (100 in this example).
```

Comment:
These docs are super valuable! But they will not be on https://api.flutter.dev/ since this is a private class. Should we move these docs up to the public class?

Reply:
Yeah, I think it is fine to include! :)

Reply:
Updated!